### PR TITLE
[ADD] V1.1 11.0_develop

### PR DESCRIPTION
**Pull request by eilst**

### CODE REVIEW

File: project_event/tests/common.py [hunk 1]
```diff
@@ -178,3 +178,41 @@ def setUp(self):
...
+            'project_id': self.project_1.id,
+            'responsible_id': self.project_1.responsible_id.id,
+            'partner_id': self.project_1.partner_id.id,
+            'category_id': self.category_1.id,
+            'room_id': self.room_1.id,
+            'spectators': '-',
+            'date_start': fields.Datetime.to_string(datetime.today()),
+            'date_end': fields.Datetime.to_string(datetime.today() +
+                                                  timedelta(hours=4)),
+        })
```
Comment (jananjoy) [inline project_event/tests/common.py]: @eilst  . You can write a test case where the time difference is only on minutes

File: project_event/models/project_task.py [hunk 2]
```diff
@@ -207,14 +212,61 @@ class Task(models.Model):
...
+            else:
+                raise ValidationError(_('Spectators value should be in the range['
+                                        + str(MIN_SPECTATORS_VALUES_LIMIT) +
+                                        '-'
+                                        + str(MAX_SPECTATORS_VALUES_LIMIT) +
+                                        ']'))
+        else:
+            self.spectators = '0'
+
+    real_date_start = fields.Datetime(
```
Comment (jananjoy) [inline project_event/models/project_task.py]: Better to match the string and variable name rel_date_start then string=' Real Date Start' or vise versa

File: project_event/models/project_task.py [hunk 3]
```diff
@@ -207,14 +212,61 @@ class Task(models.Model):
...
+        string='Actual Start Time',
+        default=None,
+    )
+    real_date_end = fields.Datetime(
+        string='Actual End Time',
+        default=None,
+    )
+    total_time = fields.Char(
```
Comment (jananjoy) [inline project_event/models/project_task.py]: @eilst better to change real or actual total time e.g real_total_time or actual_total_time

**Review by jananjoy — DISMISSED** [review]
Minor change required

File: project_event/tests/common.py (hunk 1 above)
Comment (eilst) [reply project_event/tests/common.py]: Hey hey, totally agree...but, this is your code, we merge it from
[ADD] Add section Time spent tasks (#352)this from https://github.com/savoirfairelinux/project-addons/pull/378/commits/382a162853b16f33813b8d3f338f47a7682fc3f8

File: project_event/models/project_task.py [hunk 4]
```diff
@@ -207,14 +212,64 @@ class Task(models.Model):
     )
     spectators = fields.Char(
         string='Spectators',
+        size=10,
+        default='-',
      )
```
Comment (rimbendhaou) [inline project_event/models/project_task.py]: PEP8 indentation, extra space

File: project_event/models/project_task.py [hunk 5]
```diff
@@ -207,14 +212,64 @@ class Task(models.Model):
...
+    table = fields.Char(
```
Comment (rimbendhaou) [inline project_event/models/project_task.py]: Table of what, better to have a meaningful name

File: project_event/models/project_task.py [hunk 6]
```diff
@@ -207,14 +212,64 @@ class Task(models.Model):
...
+                                        + str(MIN_SPECTATORS_VALUES_LIMIT) +
+                                        '-'
+                                        + str(MAX_SPECTATORS_VALUES_LIMIT) +
+                                        ']'))
+        else:
+            self.spectators = '0'
+
+    real_date_start = fields.Datetime(
+        string='Actual Start Time',
+        default=None,
```
Comment (rimbendhaou) [inline project_event/models/project_task.py]: by default it is already an empty field

File: project_event/models/project_task.py [hunk 7]
```diff
@@ -207,14 +212,64 @@ class Task(models.Model):
...
+    )
+    real_date_end = fields.Datetime(
+        string='Actual End Time',
+        default=None,
```
Comment (rimbendhaou) [inline project_event/models/project_task.py]: same

File: project_event/tests/common.py [hunk 8]
```diff
@@ -178,3 +178,41 @@ def setUp(self):
...
+            'name': 'Sample task3',
+            'activity_task_type': 'activity',
+            'partner_id': self.project_1.partner_id.id,
+            'room_id': self.room_1.id,
+            'date_start': fields.Datetime.to_string(datetime.today()),
+            'date_end': fields.Datetime.to_string(datetime.today() +
+                                                  timedelta(hours=4)),
+            'real_date_start': fields.Datetime.to_string(datetime.today()),
+            'real_date_end': fields.Datetime.to_string(datetime.today()),
+            'total_time': " ",
```
Comment (rimbendhaou) [inline project_event/tests/common.py]: total field is a compute field, it should be filled automatically

File: project_event/tests/common.py [hunk 9]
```diff
@@ -178,3 +178,41 @@ def setUp(self):
             'room_id': self.room_1.id,
             'notes': 'Some Task Notes ...',
         })
+        self.task_3 = self.Tasks.create({
+            'name': 'Sample task3',
```
Comment (rimbendhaou) [inline project_event/tests/common.py]: task or activity?

File: project_event/tests/common.py [hunk 10]
```diff
@@ -178,3 +178,41 @@ def setUp(self):
...
+            'activity_task_type': 'activity',
```
Comment (rimbendhaou) [inline project_event/tests/common.py]: type should be a task in this case

File: project_event/tests/common.py [hunk 11]
```diff
@@ -178,3 +178,41 @@ def setUp(self):
...
+            'date_start': fields.Datetime.to_string(datetime.today()),
+            'date_end': fields.Datetime.to_string(datetime.today() +
+                                                  timedelta(hours=4)),
+            'real_date_start': fields.Datetime.to_string(datetime.today()),
+            'real_date_end': fields.Datetime.to_string(datetime.today()),
+            'total_time': " ",
+        })
+        self.task_4 = self.Tasks.create({
+            'name': 'Sample task4',
+            'activity_task_type': 'activity',
```
Comment (rimbendhaou) [inline project_event/tests/common.py]: task or activity?

File: project_event/tests/common.py [hunk 12]
```diff
@@ -178,3 +178,41 @@ def setUp(self):
...
+            'partner_id': self.project_1.partner_id.id,
+            'room_id': self.room_1.id,
+            'date_start': fields.Datetime.to_string(datetime.today()),
+            'date_end': fields.Datetime.to_string(datetime.today() +
+                                                  timedelta(hours=4)),
+            'real_date_start': fields.Datetime.to_string(datetime.today()),
+            'real_date_end': fields.Datetime.to_string(datetime.today() +
+                                                       timedelta(hours=2)),
+            'total_time': " ",
```
Comment (rimbendhaou) [inline project_event/tests/common.py]: same comment

File: project_event/tests/common.py (hunk 11 above)
Comment (rimbendhaou) [reply project_event/tests/common.py]: Check if it is a task or activity, same comment for the above

File: project_event/views/project_task_view.xml [hunk 13]
```diff
@@ -1,5 +1,41 @@
...
+                <field name="name"/>
+                    <field name="date_start"/>
+                    <field name="real_date_start"/>
+                    <field name="date_end"/>
+                    <field name="real_date_end"/>
+                    <field name="total_time"/>
+            </tree>
+        </field>
+    </record>
+    <record id="project_event_task_tree_activitys_children" model="ir.ui.view">
```
Comment (rimbendhaou) [inline project_event/views/project_task_view.xml]: activitys ??

File: project_event/views/project_task_view.xml [hunk 14]
```diff
@@ -86,7 +112,16 @@
...
-                            <field name="spectators"/>
+                        <group>
+                            <field name="spectators" style='max-width: 900px; width: 100px'/>
+                            <field name="table" widget="widget_table" options="{
+                            'field_name': 'child_ids',
+                            'datatable_params':{'searching': False,
+                            'paging': False,
+                            'info': False},
+                            'fields': ['name','date_start', 'date_end', 'real_date_start', 'real_date_end', 'total_time'],
+                            'headers': ['Name','Date Start', 'Date End', 'Read Date Start', 'Real Date End', 'Total Time']}" />
```
Comment (rimbendhaou) [inline project_event/views/project_task_view.xml]: are these headers translatable?

File: project_event/views/project_task_view.xml [hunk 15]
```diff
@@ -1,5 +1,41 @@
...
+            </tree>
+        </field>
+    </record>
+    <record id="project_event_task_tree_activitys_children" model="ir.ui.view">
+        <field name="name">Task Tree</field>
+        <field name="model">project.task</field>
+        <field name="priority">150</field>
+        <field name="arch" type="xml">
+            <tree string="Task" default_order="date_start asc">
+                <field name="name"/>
```
Comment (rimbendhaou) [inline project_event/views/project_task_view.xml]: indentation

File: project_event/views/project_task_view.xml [hunk 16]
```diff
@@ -1,5 +1,41 @@
...
+                    <field name="equipment_id" string="Resource"/>
+                                    <field name="room_id" string="Room"/>
```
Comment (rimbendhaou) [inline project_event/views/project_task_view.xml]: indentation

**Review by rimbendhaou — CHANGES_REQUESTED** [review]

File: project_event/views/project_task_view.xml (hunk 13 above)
Comment (rimbendhaou) [inline project_event/views/project_task_view.xml]: check this tree view

File: project_event/models/project_task.py (hunk 4 above)
Comment (eilst) [reply project_event/models/project_task.py]: Done

File: project_event/models/project_task.py (hunk 5 above)
Comment (eilst) [reply project_event/models/project_task.py]: Right, done

File: project_event/models/project_task.py (hunk 7 above)
Comment (eilst) [reply project_event/models/project_task.py]: Done

File: project_event/tests/common.py (hunk 12 above)
Comment (eilst) [reply project_event/tests/common.py]: Done

File: project_event/views/project_task_view.xml (hunk 13 above)
Comment (eilst) [reply project_event/views/project_task_view.xml]: This was removed

File: project_event/views/project_task_view.xml (hunk 14 above)
Comment (eilst) [reply project_event/views/project_task_view.xml]: No, this feature is missing for the moment

File: project_event/views/project_task_view.xml (hunk 15 above)
Comment (eilst) [reply project_event/views/project_task_view.xml]: Removed

File: project_event/views/project_task_view.xml (hunk 16 above)
Comment (eilst) [reply project_event/views/project_task_view.xml]: Removed

File: project_event/views/project_task_view.xml (hunk 13 above)
Comment (jananjoy) [reply project_event/views/project_task_view.xml]: @rimbendhaou @eilst  I remove  '<' and change the typo error

File: project_event/views/project_task_view.xml (hunk 15 above)
Comment (jananjoy) [reply project_event/views/project_task_view.xml]: @rimbendhaou @eilst  Removed and  Changed

File: project_event/views/project_task_view.xml (hunk 16 above)
Comment (jananjoy) [reply project_event/views/project_task_view.xml]: @rimbendhaou @eilst  Removed and changed

File: project_event/models/project_task.py (hunk 3 above)
Comment (jananjoy) [reply project_event/models/project_task.py]: @eilst @rimbendhaou  : My change the variable name in all related files

File: project_event/views/project_task_view.xml (hunk 14 above)
Comment (jananjoy) [reply project_event/views/project_task_view.xml]: @rimbendhaou @eilst  Hope Luis is taking care in his branch.

**Review by rimbendhaou — CHANGES_REQUESTED** [review]

File: project_event/tests/common.py (hunk 11 above)
Comment (eilst) [reply project_event/tests/common.py]: You should put the fields that are only used in one test file, in that test file. Common is for fields that will be referenced in more than one test file

File: project_event/tests/test_project_event_task.py [hunk 17]
```diff
@@ -458,3 +458,48 @@ def test_190_main_task_activity_calendar_event_clone_state(self):
         for child in self.activity_1.child_ids:
             calendar_event = child.get_calendar_event()
             self.assertEqual(calendar_event.state, 'draft')
+
+    def test_200_compute_actual_total_time(self):
+        activity_vals_1 = {
+            'name': 'Sample activity 1',
```
Comment (rimbendhaou) [inline project_event/tests/test_project_event_task.py]: REal date start and real date end should be filled for a task not for an activity

File: project_event/tests/test_project_event_task.py [hunk 18]
```diff
@@ -458,3 +458,48 @@ def test_190_main_task_activity_calendar_event_clone_state(self):
...
+        activity_vals_2 = {
+            'name': 'Sample activity 2',
+            'activity_task_type': 'activity',
+            'partner_id': self.project_1.partner_id.id,
+            'room_id': self.room_1.id,
+            'date_start': fields.Datetime.to_string(datetime.today()),
+            'date_end': fields.Datetime.to_string(datetime.today() +
+                                                  timedelta(hours=4)),
+            'real_date_start': fields.Datetime.to_string(datetime.today()),
+            'real_date_end': fields.Datetime.to_string(datetime.today() +
```
Comment (rimbendhaou) [inline project_event/tests/test_project_event_task.py]: the test can fail

**Review by rimbendhaou — CHANGES_REQUESTED** [review]

File: project_event/i18n/fr.po [hunk 19]
```diff
@@ -958,6 +958,18 @@ msgstr "Séquence"
 msgid "Spectators"
 msgstr "Spectateurs"
 
+#. module: project_event
```
Comment (rimbendhaou) [inline project_event/i18n/fr.po:1088]: this translation is not used any more

File: project_event/models/project_task.py [hunk 20]
```diff
@@ -207,14 +211,67 @@ class Task(models.Model):
...
-        except Exception as exception:
-            raise ValidationError(_('The Input value should be numeric'))
+    def onchange_spectators(self):
+        if self.spectators:
+            if self.spectators == '-':
+                return
+            try:
+                valid_spectators = int(self.spectators)
+            except Exception:
+                raise ValidationError(_('Spectators value should be numeric'))
```
Comment (rimbendhaou) [inline project_event/models/project_task.py:239]: add translation for this

File: project_event/models/project_task.py [hunk 21]
```diff
@@ -207,14 +211,67 @@ class Task(models.Model):
...
+            if valid_spectators >= MIN_SPECTATORS_VALUES_LIMIT and \
+                    valid_spectators < MAX_SPECTATORS_VALUES_LIMIT:
+                self.spectators = valid_spectators
+            else:
+                raise ValidationError(
+                    _(
+                        'Spectators value should be in the range[' +
```
Comment (rimbendhaou) [inline project_event/models/project_task.py:246]: translation

File: project_event/models/project_task.py [hunk 22]
```diff
@@ -207,14 +211,67 @@ class Task(models.Model):
...
+                        self.real_date_start))
+                hours = time_diff.hours
+                minutes = time_diff.minutes
+                self.actual_total_time = "{0:0=2d}".format(
+                    hours) + ":" + "{0:0=2d}".format(minutes)
+            elif self.real_date_end == self.real_date_start:
+                self.actual_total_time = "00:00"
+            else:
+                raise ValidationError(
+                    _('Actual Start Time should be before Actual End Time'))
```
Comment (rimbendhaou) [inline project_event/models/project_task.py:272]: translation

**Review by rimbendhaou — CHANGES_REQUESTED** [review]

File: project_event/tests/test_project_event_task.py (hunk 17 above)
Comment (jananjoy) [reply project_event/tests/test_project_event_task.py]: @rimbendhaou Changed and done

File: project_event/tests/test_project_event_task.py (hunk 18 above)
Comment (jananjoy) [reply project_event/tests/test_project_event_task.py]: @rimbendhaou Changed the Code

File: project_event/models/project_task.py (hunk 20 above)
Comment (jananjoy) [reply project_event/models/project_task.py:239]: @rimbendhaou  Changed

File: project_event/tests/test_project_event_task.py [hunk 23]
```diff
@@ -459,7 +472,96 @@ def test_190_main_task_activity_calendar_event_clone_state(self):
             calendar_event = child.get_calendar_event()
             self.assertEqual(calendar_event.state, 'draft')
 
-    def test_200_get_parent_project_id(self):
+    def test_200_compute_actual_total_time(self):
+        # Test case 1: Testing for same actual start and end date
+        date_start_1 = datetime.today()
+        date_end_1 = date_start_1 + timedelta(hours=2)
+        actual_date_start_1 = datetime.today()
```
Comment (rimbendhaou) [inline project_event/tests/test_project_event_task.py]: why you use 2 variables if it is the same date?

File: project_event/tests/test_project_event_task.py [hunk 24]
```diff
@@ -459,7 +472,96 @@ def test_190_main_task_activity_calendar_event_clone_state(self):
...
+        actual_date_end_1 = actual_date_start_1
+        task_vals_1 = {
+            'name': 'Sample Task 1',
+            'activity_task_type': 'task',
+            'partner_id': self.project_1.partner_id.id,
+            'room_id': self.room_1.id,
+            'date_start': fields.Datetime.to_string(date_start_1),
+            'date_end': fields.Datetime.to_string(date_end_1),
+            'real_date_start': fields.Datetime.to_string(actual_date_start_1),
+            'real_date_end': fields.Datetime.to_string(actual_date_end_1),
```
Comment (rimbendhaou) [inline project_event/tests/test_project_event_task.py]: You can simply: 'real_date_end': fields.Datetime.to_string(actual_date_start_1),

File: project_event/tests/test_project_event_task.py (hunk 23 above)
Comment (rimbendhaou) [reply project_event/tests/test_project_event_task.py]: you can use only 1 variable :actual_date_1 = fields.Datetime.to_string(datetime.today())

File: project_event/tests/test_project_event_task.py [hunk 25]
```diff
@@ -459,7 +472,96 @@ def test_190_main_task_activity_calendar_event_clone_state(self):
...
+        }
+        task_1 = self.Tasks.create(task_vals_1)
+        diff = task_1.actual_total_time
+        self.assertEqual(diff, "00:00")
+
+        # Test case 2: Testing for time diff in hours for actual start and end date
+        date_start_2 = datetime.today()
```
Comment (rimbendhaou) [inline project_event/tests/test_project_event_task.py]: same comment. you can use the same variable for the dates

Comment (rimbendhaou) [reply project_event/tests/test_project_event_task.py]: you can also use it for all the test cases if u always have the same duration of 2hours

Comment (rimbendhaou) [reply project_event/tests/test_project_event_task.py]: these test cases can be optimized.

**Review by rimbendhaou — CHANGES_REQUESTED** [review]

File: project_event/models/project_task.py (hunk 22 above)
Comment (jananjoy) [reply project_event/models/project_task.py:272]: @rimbendhaou  Changed

File: project_event/tests/test_project_event_task.py (hunk 23 above)
Comment (jananjoy) [reply project_event/tests/test_project_event_task.py]: @rimbendhaou : To standardize for different test cases, we handle in that way. I donot think it is problem. Justify me how it improve.

File: project_event/tests/test_project_event_task.py (hunk 24 above)
Comment (jananjoy) [reply project_event/tests/test_project_event_task.py]: @rimbendhaou : That is for more clarity and readability. If the test case changes, we can change the value of the variable. if variables are reading from config files in the future, it makes sense.

File: project_event/tests/test_project_event_task.py (hunk 25 above)
Comment (jananjoy) [reply project_event/tests/test_project_event_task.py]: @rimbendhaou : Just to increase the readability of the code. Hope this is nothing to do with performance, functional behaviors etc. Anyway, you guys like to change,  No problem.

**Review by rimbendhaou — APPROVED** [review]

**Review by rimbendhaou — APPROVED** [review]

**Review by rimbendhaou — APPROVED** [review]

**Pull request merged**